### PR TITLE
Make foreach-bundle a proper shell script

### PR DESCRIPTION
**Pull request by szepeviktor**
☣️ This is highly theoretic: mostly written in GitHub editor.

### CODE REVIEW

**Comment by szepeviktor** [comment]
Should I repeat it with `fetch-org`?

**Comment by danielbachhuber** [comment]
@szepeviktor Out of curiosity, what functional benefit does this provide?

**Comment by szepeviktor** [comment]
@danielbachhuber This is way one writes Bash scripts. This is the natural way.
I try to add pointers in separate comments.

File: foreach-bundle [hunk 1]
```diff
@@ -1,20 +1,24 @@
 #!/usr/bin/env bash
+#
+# Execute a command in each subfolder.
+#
 
-BUNDLE_DEPENDENCIES=$(curl --silent https://raw.githubusercontent.com/wp-cli/wp-cli-bundle/master/composer.json -q | jq '.["require"] | keys[]' )
-BUNDLE_DEPENDENCIES=$(echo "wp-cli/wp-cli-bundle $BUNDLE_DEPENDENCIES")
+get_bundle_dependencies()
```
Comment (szepeviktor) [inline foreach-bundle:6]: - functionality goes into functions
- no variable reuse

File: foreach-bundle [hunk 2]
```diff
@@ -1,20 +1,24 @@
 #!/usr/bin/env bash
+#
+# Execute a command in each subfolder.
+#
```
Comment (szepeviktor) [inline foreach-bundle:4]: Inform the reader.

File: foreach-bundle [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
-	    if [ -d "$REPO" ]; then
-			OLDPWD="$(pwd)"
-			cd "$REPO"
-			echo "--- $REPO : ---"
-			"$@"
-			cd "$OLDPWD"
-		fi
-	esac
-done
+while read -r dependency; do
```
Comment (szepeviktor) [inline foreach-bundle:13]: pipe a function's output into a while loop, for loops are a bit more dangerous

File: foreach-bundle [hunk 4]
```diff
@@ -1,20 +1,24 @@
...
+	# Remove wp-cli/ prefix
+    repo_path="${dependency:7}"
+    if [ ! -d "${repo_path}" ]; then
+		continue
+	fi
 
+	echo "--- ${repo_path}: ---"
+	cd "${repo_path}"
+	bash -c "$@"
```
Comment (szepeviktor) [inline foreach-bundle:22]: neve use `"$@"` in itself

**Comment by danielbachhuber** [comment]
Great, thanks for clarifying @szepeviktor 😊

Let's get a second opinion from @schlessera

**Comment by schlessera** [comment]
Yes, the above sounds good to me. That script was just a quick hack I assembled and was never meant to be added as-is. But as we already have it included, we better make sure it is properly built.

**Review by schlessera — APPROVED** [review]

**Comment by szepeviktor** [comment]
Thank you.

**Pull request merged**